services/cargohold/src/CargoHold/API/Public.hs

```diff
@@ -136,9 +143,28 @@ uploadAssetV3 ::
   AssetSource ->
   Handler (Asset, AssetLocation Relative)
 uploadAssetV3 pid req = do
+  clientEnv <- asks $ view brigClientEnv
   let principal = mkPrincipal pid
+      getUser uid = runClientM (client' uid) clientEnv
+  case principal of
+    V3.UserPrincipal uid -> do
+      resp <- liftIO $ getUser uid
+      users <- either (const $ throwE userNotFound) pure resp
+      maybe
+        (throwE unverified)
+        (const (pure ()) . userIdentity . accountUser)
+        (listToMaybe users)
+    _ -> pure ()
   asset <- V3.upload principal (getAssetSource req)
   pure (fmap tUntagged asset, mkAssetLocation @tag (asset ^. assetKey))
+  where
+    client' uid =
```

Comment:
```suggestion
    getUserById uid =
```

It  might even make sense to just move the whole `runClientM` and `clientM` thing in here. Perhaps even start createing an "internal  brig client" in `wire-api`, so we don't have to do this adhoc servant stuff.

Reply:
Done, and I've moved the `runClientM` around.